pkg/api/types.go

```diff
@@ -1045,7 +1045,10 @@ func (p *Properties) IsUbuntuDistroForAllNodes() bool {
 			}
 		}
 	}
-	return p.MasterProfile.Distro == Ubuntu || p.MasterProfile.Distro == Ubuntu1804
+	if p.MasterProfile != nil {
```

Comment:
Good catch. Do we have a unit test case that covers the nil p.MasterProfile scenario?